apis/python/src/tiledb/vector_search/ingestion.py

```diff
@@ -1795,41 +1783,25 @@ def ingest_vectors_udf(
                     trace_id=trace_id,
                 )
                 logger.debug("Start indexing")
-                if index_timestamp is None:
-                    ivf_index(
-                        dtype=vector_type,
-                        db=array_to_matrix(
-                            np.transpose(in_vectors).astype(vector_type)
-                        ),
-                        external_ids=StdVector_u64(external_ids),
-                        deleted_ids=StdVector_u64(updated_ids),
-                        centroids_uri=centroids_uri,
-                        parts_uri=partial_write_array_parts_uri,
-                        index_array_uri=partial_write_array_index_uri,
-                        id_uri=partial_write_array_ids_uri,
-                        start=part,
-                        end=part_end,
-                        nthreads=threads,
-                        config=config,
-                    )
-                else:
-                    ivf_index(
-                        dtype=vector_type,
-                        db=array_to_matrix(
-                            np.transpose(in_vectors).astype(vector_type)
-                        ),
-                        external_ids=StdVector_u64(external_ids),
-                        deleted_ids=StdVector_u64(updated_ids),
-                        centroids_uri=centroids_uri,
-                        parts_uri=partial_write_array_parts_uri,
-                        index_array_uri=partial_write_array_index_uri,
-                        id_uri=partial_write_array_ids_uri,
-                        start=part,
-                        end=part_end,
-                        nthreads=threads,
-                        timestamp=index_timestamp,
-                        config=config,
-                    )
+                ivf_index(
+                    dtype=vector_type,
+                    db=array_to_matrix(np.transpose(in_vectors).astype(vector_type)),
+                    external_ids=StdVector_u64(external_ids),
+                    deleted_ids=StdVector_u64(updated_ids),
+                    centroids_uri=centroids_uri,
+                    parts_uri=partial_write_array_parts_uri,
+                    index_array_uri=partial_write_array_index_uri,
+                    id_uri=partial_write_array_ids_uri,
+                    start=part,
+                    end=part_end,
+                    nthreads=threads,
+                    **(
+                        {"timestamp": index_timestamp}
+                        if index_timestamp is not None
+                        else {}
+                    ),
```

Comment:
Just to share, beyond normal testing, I also tested this with a small example:
```
def foo(cool, timestamp = 1):
    print('timestamp', timestamp)

def test_foo():
    print()

    foo('a', timestamp=0)
    
    foo('a', 0)

    index_timestamp = None
    foo(
        'a', 
        **({"timestamp": index_timestamp} if index_timestamp is not None else {}
    )
    )

    index_timestamp = 2
    foo(
        'a', 
        **({"timestamp": index_timestamp} if index_timestamp is not None else {}
    )
    )
```
which prints:
```
timestamp 0
timestamp 0
timestamp 1
timestamp 2
```